packages/core/src/external/commonwell/document/document-downloader-local.ts

```diff
@@ -50,13 +51,34 @@ export class DocumentDownloaderLocal extends DocumentDownloader {
     const onEnd = () => {
       console.log("Finished downloading document");
     };
-    const downloadResult = await this.downloadFromCommonwellIntoS3(
-      document,
-      fileInfo,
-      onData,
-      onEnd
-    );
+    let downloadResult = await this.downloadFromCommonwellIntoS3(document, fileInfo, onData, onEnd);
 
+    // Check if the detected file type is in the accepted content types
+    if (!isContentTypeAccepted(document.mimeType)) {
+      // If not, update the content type in S3
+      const [detectedFileType, detectedExtension] = detectFileType(Buffer.from(downloadedDocument));
```

Comment:
If not a lot of effort, we could try to create a buffer from just the first few bytes that `detectFileType` needs, so Buffer doesn't duplicate the memory used by `downloadedDocument`.

Alternatively, could store those on the buffer as we load the file on line 49.

Reply:
Only consideration is that checking if its a text file requires the whole buffer. I made a small refactor so that it will only pass in the whole buffer if we can't use the magic numbers approach to easily determine the file type